### PR TITLE
game-action/minetest: Version bump to 0.4.13

### PR DESCRIPTION
**Pull request by damiel**
As promised a cleaned up PR with the most recent version.

I have to mention tho that the doc doesn't get installed properly and i wasn't able to track down why it fails to do that. Stepping through the ebuild with the ebuild command only showed that the doc is getting generated but not installed.

### CODE REVIEW

File: games-action/minetest/files/minetest-0.4.13-shared-irrlicht.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
+diff --git a/cmake/Modules/FindIrrlicht.cmake b/cmake/Modules/FindIrrlicht.cmake
```
Comment (hasufell) [inline games-action/minetest/files/minetest-0.4.13-shared-irrlicht.patch]: missing patch description

**Comment by hasufell** [comment]
> I have to mention tho that the doc doesn't get installed properly and i wasn't able to track down why it fails to do that.

because portage is broken, it works fine here under paludis (if you mean the doxygen documentation)

**Comment by hasufell** [comment]
automagic dependency on `sci-libs/libspatialindex`

File: games-action/minetest/minetest-0.4.13.ebuild [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		$(cmake-utils_use_enable truetype FREETYPE)
+		$(cmake-utils_use_enable nls GETTEXT)
+		-DENABLE_GLES=0
+		$(cmake-utils_use_enable leveldb LEVELDB)
+		$(cmake-utils_use_enable redis REDIS)
+		$(cmake-utils_use_enable sound SOUND)
+		-DDISABLE_LUAJIT=$(usex !luajit)
+		-DRUN_IN_PLACE=0
+	)
+
```
Comment (hasufell) [inline games-action/minetest/minetest-0.4.13.ebuild]: unnecessary whitespace here

**Comment by hasufell** [comment]
`games-action/minetest_game` is a different package and not a direct dependency, so it can and should be committed AFTER `games-action/minetest`.

**Comment by damiel** [comment]
Yes i am meaning the doxygen documentation.

**Comment by hasufell** [comment]
You didn't fix the dependencies mentioned in https://github.com/gentoo/gentoo/pull/28#issuecomment-133569524

**Comment by damiel** [comment]
I am still working on it :P

File: games-action/minetest/metadata.xml [hunk 3]
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-	<herd>games</herd>
```
Comment (hasufell) [inline games-action/minetest/metadata.xml]: We don't just remove maintainers/herds, unless there is a particular reason for it.

Comment (damiel) [inline games-action/minetest/metadata.xml]: Is the games herd being a pity illusion of itself not reason enough to drop it? Games herd did nothing for this ebuild and i am the sole maintainer of it and i as a maintainer decide that i don't need nor want the games herd.

Comment (hasufell) [inline games-action/minetest/metadata.xml]: The primary maintainer is the games herd. I was the previous maintainer (and importer) and I worked on this ebuild with the help of the games herd.

I am well (more then you think) aware of the problems with the games herd (I am not one of them). But just removing them from metadata.xml files is not a clean solution. Decisions about maintainership are not up to proxy-maintainers.

Comment (damiel) [inline games-action/minetest/metadata.xml]: I know that you were the previous maintainer, i followed every step of the minetest package since its very beginning in your own public overlay over its import to the portage tree till now when i got the (proxy-)maintainer.

It is already agreed on that this ebuild is out of the games herd responsibility as you can read for example at the first comment by @mgorny in my last PR #17.

Comment (hasufell) [inline games-action/minetest/metadata.xml]: That was a false statement from @mgorny. Except for him, there is no consensus on agreeing to ignore them. They are still reviewing ebuilds, see https://github.com/gentoo/gentoo/commit/b964bb05fe54c61122e225a623d6349f11c7363b

A clean solution to this involves a ComRel action or even the council. None of this has to be discussed in this PR.

Comment (damiel) [inline games-action/minetest/metadata.xml]: I agree on your last paragraph, this is out of the scope of this PR.

Comment (hasufell) [inline games-action/minetest/metadata.xml]: > I agree on your last paragraph, this is out of the scope of this PR.

Then you should know that this will not be merged with that maintainer removal.

Comment (damiel) [inline games-action/minetest/metadata.xml]: I can wait till the decision is done.

Comment (hasufell) [inline games-action/minetest/metadata.xml]: > but i hope that this issue gets fixed soon, otherwise i don't see how reasonable work is possible

We are working on it.

Anyway, since you are proxy-maintainer, I am happy to merge this if you keep the `metadata.xml` as-is. Also mind that all of this does not apply if you import a new ebuild. Then you are not obliged to add the gamers herd.

Comment (damiel) [inline games-action/minetest/metadata.xml]: Can you point out please where the issue with sci-libs/libspatialindex lies? I cant find any reference to it and i dont even have it installed on my system and yet everything seems to compile just fine.

Comment (hasufell) [inline games-action/minetest/metadata.xml]: > Can you point out please where the issue with sci-libs/libspatialindex lies?

https://github.com/minetest/minetest/blob/master/src/CMakeLists.txt#L183

It just silently disables the feature if it is not found, hence we get automagic linking.

Comment (damiel) [inline games-action/minetest/metadata.xml]: Isn't that worth to patch the build system(yet again) or do we just add the dependency?

Comment (hasufell) [inline games-action/minetest/metadata.xml]: > Isn't that worth it to patch the build system(yet again) or is it sufficient to add the dependency?

It is already optional (via `-DENABLE_SPATIAL=<ON/OFF>`). So it could be made either
* a USE flag (with the description "Enable SpatialIndex AreaStore backend")
* just enable it unconditionally

Comment (damiel) [inline games-action/minetest/metadata.xml]: I am asking right now at the minetest-dev channel in irc what implifications it might have for users but id rather have it as a useflag that can be enable by users then just enabling it without people knowing that it is there.

Comment (hasufell) [inline games-action/minetest/metadata.xml]: > How would you call the useflag, is "spatial" good enough?

probably (as that's how the build system calls it too)

**Comment by damiel** [comment]
Thanks for your input @hasufell.

**Comment by damiel** [comment]
Anything more that could be improved?

**Comment by hasufell** [comment]
yeah, squash the commits

**Comment by hasufell** [comment]
> How about for example the doxygen doc issue, anything i can do there?

* stop using portage
or
* don't use `dohtml` and try with `doins` instead

**Comment by mgorny** [comment]
Oh yes, please don't use `dohtml`. It is deprecated.

**Comment by mgorny** [comment]
@hasufell, not `doins`, `docinto` and `dodoc`.

**Comment by hasufell** [comment]
> not doins, docinto and dodoc.

pretty much the same

**Comment by damiel** [comment]
The ebuild uses dodoc, but ill look into docinto to see if that a better option

File: games-action/minetest/files/minetest-0.4.13-shared-irrlicht.patch (hunk 1 above)
Comment (hasufell) [inline games-action/minetest/files/minetest-0.4.13-shared-irrlicht.patch]: as I see it, this patch is obsolete

**Comment by damiel** [comment]
Shall i squash all the commits into one or shall i keep games-action/minetest and games-action/minetest_games seperated(therefore creating one commit for games-action/minetest and one for games-action/minetest_game)?

**Comment by mgorny** [comment]
USE=doc installs the docs correctly for me using Portage. Are you sure you didn't try with USE=-doc?

But there's another file that gets installed into incorrect location:

```
--- /usr/share/doc/minetest/
>>> /usr/share/doc/minetest/minetest.conf.example.lz

**Comment by damiel** [comment]
Yes i am sure that i have the flag set properly.

**Comment by damiel** [comment]
> But there's another file that gets installed into incorrect location:

Trying to override its installation target with -DCUSTOM_CUSTOM_EXAMPLE_CONF_DIR="/usr/share/doc/${PF}" failed for me.

**Comment by damiel** [comment]
I fixed the wrong install location of minetest.conf.example and squashed the commits once again.

**Comment by damiel** [comment]
There is a bug filed now on bugzilla, i drop it here as reference: https://bugs.gentoo.org/show_bug.cgi?id=558978

File: games-action/minetest/minetest-0.4.13.ebuild [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+		-DCUSTOM_EXAMPLE_CONF_DIR="/usr/share/doc/${PF}"
+		-DENABLE_CURL=$(usex curl)
+		$(cmake-utils_use_enable truetype FREETYPE)
+		$(cmake-utils_use_enable nls GETTEXT)
+		-DENABLE_GLES=0
+		$(cmake-utils_use_enable leveldb LEVELDB)
+		$(cmake-utils_use_enable redis REDIS)
+		-DENABLE_SPATIAL=$(usex spatial)
+		$(cmake-utils_use_enable sound SOUND)
+		-DDISABLE_LUAJIT=$(usex !luajit)
```
Comment (hasufell) [inline games-action/minetest/minetest-0.4.13.ebuild]: `$(cmake-utils_use luajit ENABLE_LUAJIT)`

**Comment by damiel** [comment]
Any opinions on opengles and luajit? I posted mine on the bugreport.

**Comment by mgorny** [comment]
One more issue; it bundles fonts:

```
/usr/share/minetest/fonts/liberationsans.ttf
/usr/share/minetest/fonts/liberationmono.ttf
/usr/share/minetest/fonts/DroidSansFallbackFull.ttf
```

Since it's minor, we have two options now:
* you remove the bundled fonts. Either make sure that the game can use global fonts and add appropriate deps, or symlink them.
* you leave the bundled fonts with a bug open to clean them up in the future, but you have to add appropriate licenses in `LICENSE` for the ebuild.
